presto-parquet/src/test/java/com/facebook/presto/parquet/TestParquetFileStats.java

```diff
@@ -0,0 +1,206 @@
+                minSlice = slice;
+            }
+        }
+        // thrown assert-error when using parquet-mr prior 1.10.0 (PARQUET-1025)
+        assertEquals(minSlice, Slices.utf8Slice(min), "Min-binary and Min-slice mismatch");
+        assertEquals(maxSlice, Slices.utf8Slice(max), "Max-binary and Max-slice mismatch");
+    }
+
+    @Test
+    public void testParquetFile()
+            throws Exception
+    {
+        testParquetFile(testStrings, false);
+        testParquetFile(testStrings, true);
+    }
+
+    public void testParquetFile(List<String> strings, boolean containEmptyString)
+            throws Exception
+    {
+        if (containEmptyString) {
+            strings.add("");
+        }
+        Configuration conf = new Configuration();
+        Path root = new Path("target/tests/TestParquetWriter/");
+        enforceEmptyDir(conf, root);
+        MessageType schema = parseMessageType(
+                "message test { "
+                        + "required int32 int32_field; "
+                        + "required binary binary_field; "
+                        + "required fixed_len_byte_array(3) flba_field; "
+                        + "} ");
+        GroupWriteSupport.setSchema(schema, conf);
+        SimpleGroupFactory f = new SimpleGroupFactory(schema);
+        Map<String, Encoding> expected = new HashMap<String, Encoding>();
+        expected.put("10-" + PARQUET_1_0, PLAIN_DICTIONARY);
+        expected.put("1000-" + PARQUET_1_0, PLAIN);
+        expected.put("10-" + PARQUET_2_0, RLE_DICTIONARY);
+        expected.put("1000-" + PARQUET_2_0, DELTA_BYTE_ARRAY);
+        for (int modulo : asList(10, 1000)) {
+            for (ParquetProperties.WriterVersion version : ParquetProperties.WriterVersion.values()) {
+                Path file = new Path(root, version.name() + "_" + modulo);
+                ParquetWriter<Group> writer = new ParquetWriter<Group>(
+                        file,
+                        new GroupWriteSupport(),
+                        UNCOMPRESSED, 1024, 1024, 512, true, false, version, conf);
+                for (int i = 0; i < strings.size(); i++) {
+                    for (int j = 0; j < 1000; j++) {
+                        writer.write(f.newGroup()
+                                .append("int32_field", i)
+                                .append("binary_field", strings.get(i) + j % modulo)
+                                .append("flba_field", "foo"));
+                    }
+                }
+
+                writer.close();
+                ParquetReader<Group> reader = ParquetReader.builder(new GroupReadSupport(), file).withConf(conf).build();
+                for (int i = 0; i < strings.size(); i++) {
+                    for (int j = 0; j < 1000; j++) {
+                        Group group = reader.read();
+                        assertEquals(strings.get(i) + j % modulo, group.getBinary("binary_field", 0).toStringUsingUTF8());
```

Comment:
having `strings.get(i) + j % modulo` hard-coded in two places is error-prone; consider generating data first, then calling helper method to write and read that data

```
            ImmutableList.Builder<Integer> intValuesBuilder = ImmutableList.builder();
            ImmutableList.Builder<String> stringValuesBuilder = ImmutableList.builder();
            for (int i = 0; i < strings.size(); i++) {
                for (int j = 0; j < 1000; j++) {
                    intValuesBuilder.add(i);
                    stringValuesBuilder.add(strings.get(i) + j % modulo);
                }
            }

            List<Integer> intValues = intValuesBuilder.build();
            List<String> stringValues = stringValuesBuilder.build();

            for (ParquetProperties.WriterVersion version : ParquetProperties.WriterVersion.values()) {
                Path file = new Path(rootDirectory, version.name() + "_" + modulo);
                writeValues(file, version, schema, conf, intValues, stringValues);
                assertFileContents(file, conf, intValues, stringValues);
...
    private void writeValues(Path file, ParquetProperties.WriterVersion version, MessageType schema, Configuration conf, List<Integer> intValues, List<String> stringValues)
            throws IOException
    {
        ParquetWriter<Group> writer = new ParquetWriter<>(
                file,
                new GroupWriteSupport(),
                UNCOMPRESSED, 1024, 1024, 512, true, false, version, conf);
        SimpleGroupFactory factory = new SimpleGroupFactory(schema);
        for (int i = 0; i < intValues.size(); i++) {
            writer.write(factory.newGroup()
                    .append("int32_field", intValues.get(i))
                    .append("binary_field", stringValues.get(i))
                    .append("flba_field", "foo"));
        }
        writer.close();
    }

    private void assertFileContents(Path file, Configuration conf, List<Integer> intValues, List<String> stringValues)
            throws IOException
    {
        ParquetReader<Group> reader = ParquetReader.builder(new GroupReadSupport(), file).withConf(conf).build();
        for (int i = 0; i < intValues.size(); i++) {
            Group group = reader.read();
            assertEquals(intValues.get(i).intValue(), group.getInteger("int32_field", 0));
            assertEquals(stringValues.get(i), group.getBinary("binary_field", 0).toStringUsingUTF8());
        }
        reader.close();
    }
```

Reply:
@mbasmanova I'm not sure if it's necessary to add parquet's stats production logic.If not, class TestParquetFileStats may be removed